<a href="https://colab.research.google.com/github/nnekandubishi/netflix-personalised-dashboard/blob/main/netflix_personalised_dashboard_git.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Connecting GitHub to Colab**

In [1]:
!git clone https://github.com/nnekandubishi/netflix-personalised-dashboard.git


Cloning into 'netflix-personalised-dashboard'...
remote: Enumerating objects: 31, done.
remote: Counting objects: 100% (31/31), done.
remote: Compressing objects: 100% (29/29), done.
remote: Total 31 (delta 10), reused 0 (delta 0), pack-reused 0 (from 0)
Receiving objects: 100% (31/31), 51.31 KiB | 2.05 MiB/s, done.
Resolving deltas: 100% (10/10), done.


In [2]:
%cd netflix-personalised-dashboard/data-cleaning

/content/netflix-personalised-dashboard/data-cleaning


In [3]:
# Check current directory and list files
!pwd
!ls

/content/netflix-personalised-dashboard/data-cleaning
cleaning_notes.py  README.md


In [4]:
%cd ../

/content/netflix-personalised-dashboard


In [5]:
!ls

data-cleaning  LICENSE	netflix_viewing_data.csv  README.md


In [6]:
%cd netflix-personalised-dashboard

[Errno 2] No such file or directory: 'netflix-personalised-dashboard'
/content/netflix-personalised-dashboard


In [7]:
import pandas as pd

df = pd.read_csv('netflix_viewing_data.csv', encoding='ISO-8859-1')

print(df.head())

                Title    Genre        Director      Country  Release Year  \
0  You are My Destiny  Romance  Kim Myung-wook  South Korea        2014.0   
1      Love Next Door  Romance             NaN  South Korea           NaN   
2      Love Next Door  Romance             NaN  South Korea           NaN   
3      Love Next Door  Romance             NaN  South Korea           NaN   
4      Love Next Door  Romance             NaN  South Korea           NaN   

  Content Type     Duration Rating  IMDb Rating Language Date Watched  
0    TV Series  36 episodes  PG-13          7.1   Korean   24/09/2024  
1    TV Series          NaN    NaN          NaN   Korean   22/09/2024  
2    TV Series          NaN    NaN          NaN   Korean   22/09/2024  
3    TV Series          NaN    NaN          NaN   Korean   22/09/2024  
4    TV Series          NaN    NaN          NaN   Korean   22/09/2024  


# **Data Cleaning**

## **1: Treating NULL Values**

### 1.1 Checking for NULL Values:

In [8]:
print(df.isnull().sum())

Title             0
Genre            49
Director        129
Country          58
Release Year     89
Content Type     59
Duration        109
Rating          192
IMDb Rating     118
Language         58
Date Watched      0
dtype: int64


### 1.2 Handling NULL Values:

**1.2.1: Director**

The Director column has 129 missing values. Missing values will be filled with "Unknown" as a placeholder to ensure completeness of the dataset.

In [9]:
df['Director'].fillna('Unknown', inplace=True)

<ipython-input-9-0c904c39178e>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Director'].fillna('Unknown', inplace=True)


**1.2.2: Genre**

The Genre column has 49 missing values. Missing values in the Genre column will be filled with "Unknown" to account for any unavailable data.

In [10]:
df['Genre'].fillna('Unknown', inplace=True)

<ipython-input-10-f064f74bf2d5>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Genre'].fillna('Unknown', inplace=True)


**1.2.3: Country**

The Country column has 58 missing values. The missing values in this column will be replaced with "Unknown" to indicate that the country of origin is not specified.

In [11]:
df['Country'].fillna('Unknown', inplace=True)

<ipython-input-11-16920c969032>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Country'].fillna('Unknown', inplace=True)


**1.2.4: Release Year**

The Release Year column has 89 missing values. We will fill these missing values with "0" to represent cases where the release year is not available.

In [12]:
df['Release Year'].fillna(0, inplace=True)

<ipython-input-12-849c23cb9899>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Release Year'].fillna(0, inplace=True)


**1.2.5: Content Type**

The Content Type column has 59 missing values. Missing values will be filled with "Unknown" to represent cases where the content type (e.g., movie or TV show) is not available.

In [13]:
df['Content Type'].fillna('Unknown', inplace=True)

<ipython-input-13-bde3eaf9992e>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Content Type'].fillna('Unknown', inplace=True)


**1.2.6: Duration**

The Duration column has 109 missing values. Missing values in this column will be filled with "Unknown" as a placeholder for cases where the duration is unavailable.

In [14]:
df['Duration'].fillna('Unknown', inplace=True)

<ipython-input-14-18ff8f47c08e>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Duration'].fillna('Unknown', inplace=True)


**1.2.7: Rating**

The Rating column has 192 missing values. Missing ratings will be filled with "Not Rated" to indicate that no rating information is available for these entries.

In [15]:
df['Rating'].fillna('Not Rated', inplace=True)

<ipython-input-15-4bf1dc6bb789>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Rating'].fillna('Not Rated', inplace=True)


**1.2.8: IMDb Rating**

The IMDb Rating column has 118 missing values. These missing values will be filled with "0" to represent cases where IMDb ratings are unavailable.

In [16]:
df['IMDb Rating'].fillna(0, inplace=True)

<ipython-input-16-272caa274cdd>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['IMDb Rating'].fillna(0, inplace=True)


**1.2.9: Language**

The Language column has 58 missing values. Missing values will be filled with "Unknown" to account for cases where the language information is not specified.

In [17]:
df['Language'].fillna('Unknown', inplace=True)

<ipython-input-17-193ed3a653cf>:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Language'].fillna('Unknown', inplace=True)


## **2. Unique Entries in the Country Column**

### **Identifying all the entries in the Country Column**

In [18]:
# Display unique country entries
unique_countries = df['Country'].unique()
print(unique_countries)

['South Korea' 'Japan' 'Unknown' 'China' 'USA' 'Spain' 'France' 'UK'
 'Belgium' 'USA, Poland' 'Nigeria' 'India' 'UK, USA' 'South Africa'
 'Kenya' 'Ireland, Canada' 'Turkey' 'Czech Republic' 'USA, UK' 'Germany'
 'USA, Australia' 'Poland' 'USA, Canada' 'New Zealand' 'Norway' 'Mexico'
 'Ghana' 'Canada' 'Italy' 'Russia' 'Hong Kong' 'Vietnam' 'Philippines'
 'Australia' 'Colombia']


### **Split Multiple Countries**
Keeping only the first country mentioned:

In [19]:
# Split by commas and keep the first country
df['Country'] = df['Country'].apply(lambda x: x.split(',')[0].strip() if ',' in x else x)


In [20]:
# Verify the changes
cleaned_unique_countries = df['Country'].unique()
print(cleaned_unique_countries)

['South Korea' 'Japan' 'Unknown' 'China' 'USA' 'Spain' 'France' 'UK'
 'Belgium' 'Nigeria' 'India' 'South Africa' 'Kenya' 'Ireland' 'Turkey'
 'Czech Republic' 'Germany' 'Poland' 'New Zealand' 'Norway' 'Mexico'
 'Ghana' 'Canada' 'Italy' 'Russia' 'Hong Kong' 'Vietnam' 'Philippines'
 'Australia' 'Colombia']


# **DataFrame Creation for Informational Analysis**

To ensure that analyses such as genre distribution, IMDb ratings, and content type breakdowns are not affected by duplicate entries, we created a new DataFrame, df_info_only. This DataFrame focuses solely on the informational aspects of the dataset and excludes any time-based data.

**Key steps:**

Dropped the 'Date Watched' column: Since time-based analysis is not relevant for this DataFrame, the Date Watched column was removed.

Removed duplicates based on 'Title': To avoid multiple entries of the same show or movie, duplicates were removed, keeping only the first occurrence of each unique title.

This cleaned DataFrame will be used for various analyses, including genre, ratings, and content type distribution, without the influence of duplicate records.

In [21]:
df_info_only = df.drop(columns=['Date Watched']).drop_duplicates(subset=['Title'])

print(df_info_only.head())

                Title     Genre        Director      Country  Release Year  \
0  You are My Destiny   Romance  Kim Myung-wook  South Korea        2014.0   
1      Love Next Door   Romance         Unknown  South Korea           0.0   
5                 Eve  Thriller  Park Bong-seop  South Korea        2022.0   
8       Chastity High   Unknown         Unknown  South Korea           0.0   
9      Hotel Del Luna   Fantasy  Oh Choong-hwan  South Korea        2019.0   

  Content Type     Duration     Rating  IMDb Rating Language  
0    TV Series  36 episodes      PG-13          7.1   Korean  
1    TV Series      Unknown  Not Rated          0.0   Korean  
5    TV Series  16 episodes        18+          7.5   Korean  
8    TV Series      Unknown  Not Rated          0.0   Korean  
9    TV Series  16 episodes      PG-13          8.2   Korean  


# **1. Time-Based Analysis**

**Summary:**

Most Watched Month: Shows the month with the highest viewing activity.

Least Watched Month: Shows the month with the lowest viewing activity.

Most Watched Year: Shows the year with the most content watched.

Least Watched Year: Shows the year with the least content watched.

### **1.1 Extract the Month and Year**

Extracting the month and year from the Date Watched column to group the content watched by month.

In [22]:
# Rename df to df_time_series
df_time_series = df.copy()

In [23]:
# Convert 'Date Watched' to datetime format
df_time_series['Date Watched'] = pd.to_datetime(df_time_series['Date Watched'], format='%d/%m/%Y', errors='coerce')

In [24]:
# Extract month and year from 'Date Watched'
df_time_series['Month'] = df_time_series['Date Watched'].dt.month
df_time_series['Year'] = df_time_series['Date Watched'].dt.year

In [25]:
# Group by Year and Month and count the number of shows/movies watched
content_per_month = df_time_series.groupby(['Year', 'Month']).size()

content_per_month

Year  Month
2017  8         2
      9        45
      10       92
      11       62
      12       51
               ..
2024  5        58
      6        21
      7        59
      8        43
      9        17
Length: 86, dtype: int64

### **1.2 Find the month with the Most and Least Viewing Activity**

In [26]:
most_watched_month = content_per_month.idxmax()
most_watched_count = content_per_month.max()

least_watched_month = content_per_month.idxmin()
least_watched_count = content_per_month.min()

print(f"The most watched month: {most_watched_month} with {most_watched_count} shows/movies watched.")
print(f"The least watched month: {least_watched_month} with {least_watched_count} shows/movies watched.")


The most watched month: (2021, 1) with 157 shows/movies watched.
The least watched month: (2017, 8) with 2 shows/movies watched.


### **1.3 Find the Year with the Most and Least Viewing Activity**

In [27]:
content_per_year = df_time_series.groupby('Year').size()

most_watched_year = content_per_year.idxmax()
most_watched_year_count = content_per_year.max()

least_watched_year = content_per_year.idxmin()
least_watched_year_count = content_per_year.min()

print(f"The year with the most viewing activity: {most_watched_year} with {most_watched_year_count} shows/movies watched.")
print(f"The year with the least viewing activity: {least_watched_year} with {least_watched_year_count} shows/movies watched.")

The year with the most viewing activity: 2021 with 1008 shows/movies watched.
The year with the least viewing activity: 2017 with 252 shows/movies watched.


# **2. Genre Analysis**

**Summary:**

Genre Count: Shows the total number of shows/movies watched for each genre.

Most/Least Watched Genres: Identifies which genres you watched the most and least.

Highest/Lowest Rated Genres: Compares IMDb ratings across genres.

Listing the shows with IMDb ratings higher than 8.0 with their genres.

## **Count of All Genres**
This gives a full breakdown of how many shows or movies fall under each genre.

In [28]:
genre_count = df_info_only['Genre'].value_counts()

print("Count of all genres:")
genre_count

Count of all genres:


Genre
Romance        130
Action         121
Drama          113
Comedy          97
Fantasy         77
Thriller        68
Sci-Fi          44
Crime           42
Historical      19
Animation       18
Documentary     17
Reality         14
Mystery         14
Unknown         12
Adventure       10
Horror           7
Family           6
Legal            5
Medical          2
Biography        2
Sports           1
Political        1
Western          1
War              1
Talk Show        1
Name: count, dtype: int64

## **Most and Least Watched Genres**
This identifies what I've watched the most and the least.

In [29]:
most_watched_genre = genre_count.idxmax()
most_watched_genre_count = genre_count.max()

least_watched_genre = genre_count.idxmin()
least_watched_genre_count = genre_count.min()

print(f"\nThe most watched genre: {most_watched_genre} with {most_watched_genre_count} shows/movies watched.")
print(f"The least watched genre: {least_watched_genre} with {least_watched_genre_count} shows/movies watched.")


The most watched genre: Romance with 130 shows/movies watched.
The least watched genre: Sports with 1 shows/movies watched.


## **Genres with Highest and Lowest IMDb Ratings (Excluding 0 Ratings)**


In [30]:
# Exclude rows where IMDb Rating is 0 (indicating unknown or not rated)
df_rated = df_info_only[df_info_only['IMDb Rating'] != 0]

# Convert 'IMDb Rating' to numeric if needed
df_rated['IMDb Rating'] = pd.to_numeric(df_rated['IMDb Rating'], errors='coerce')

# Group by Genre and calculate the mean IMDb Rating for each genre
imdb_by_genre = df_rated.groupby('Genre')['IMDb Rating'].mean()

# Find the genre with the highest IMDb rating
highest_rated_genre = imdb_by_genre.idxmax()
highest_imdb_rating = imdb_by_genre.max()

# Find the genre with the lowest IMDb rating
lowest_rated_genre = imdb_by_genre.idxmin()
lowest_imdb_rating = imdb_by_genre.min()

print(f"\nThe genre with the highest IMDb rating: {highest_rated_genre} with an average rating of {highest_imdb_rating}.")
print(f"The genre with the lowest IMDb rating: {lowest_rated_genre} with an average rating of {lowest_imdb_rating}.")


The genre with the highest IMDb rating: Sports with an average rating of 8.4.
The genre with the lowest IMDb rating: Reality with an average rating of 6.385714285714286.


<ipython-input-30-7d54abf45934>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_rated['IMDb Rating'] = pd.to_numeric(df_rated['IMDb Rating'], errors='coerce')


## **TV Series with Romance Genre and IMDb Rating > 8.0:**

In [31]:
high_rated_romance_tv = df_info_only[(df_info_only['IMDb Rating'] > 8.0) &
                                     (df_info_only['Genre'].str.contains('Romance')) &
                                     (df_info_only['Content Type'] == 'TV Series')]
high_rated_romance_tv[['Title', 'Genre', 'IMDb Rating']]

Title    Genre  IMDb Rating
62          Twenty Five Twenty One  Romance          8.7
73                   Coffee Prince  Romance          8.2
176             Suspicious Partner  Romance          8.1
290                   Youth of May  Romance          8.5
317               Fight for My Way  Romance          8.1
364          Something in the Rain  Romance          8.1
439           Hometown Cha-Cha-Cha  Romance          8.4
475  Because This Is My First Life  Romance          8.1
490          My Love from the Star  Romance          8.2
537          Strong Girl Bong-soon  Romance          8.2
566                    Hidden Love  Romance          8.8
594           Crash Landing on You  Romance          8.7
646         Descendants of the Sun  Romance          8.3
671             Our Beloved Summer  Romance          8.4
674       It's Okay to Not Be Okay  Romance          8.6
753              Business Proposal  Romance          8.1

## **TV Series with Crime Genre and IMDb Rating > 8.0:**

In [32]:
high_rated_crime_tv = df_info_only[(df_info_only['IMDb Rating'] > 8.0) &
                                   (df_info_only['Genre'].str.contains('Crime')) &
                                   (df_info_only['Content Type'] == 'TV Series')]
high_rated_crime_tv[['Title', 'Genre', 'IMDb Rating']]

Title  Genre  IMDb Rating
529          Money Heist  Crime          8.2
613               Luther  Crime          8.4
760              Top Boy  Crime          8.4
1070               Power  Crime          8.1
1310      Peaky Blinders  Crime          8.8
1633               Ozark  Crime          8.5
1889              Narcos  Crime          8.8
3039            The Fall  Crime          8.1
3103      Ashes to Ashes  Crime          8.2
3131        White Collar  Crime          8.2
3581        Line of Duty  Crime          8.7
3597    When They See Us  Crime          8.9
4152  Person of Interest  Crime          8.4
4683          MINDHUNTER  Crime          8.6
4779              Dexter  Crime          8.7

# **3. Ratings Analysis**

**Summary:**

IMDb Ratings Distribution: Provides an overview of how ratings are spread across the content you've watched, showing the frequency of each IMDb rating.

Ratings by Content Type: Calculates the average IMDb rating for different content types (e.g., movies or TV series) to identify any trends.

## **IMDb Ratings Distribution**
Analysing the overall distribution of IMDb ratings for the entire dataset (excluding any 0 ratings that represent unrated content). This gives us insight into how content is rated in general.

In [33]:
# Exclude rows where IMDb Rating is 0 (indicating unknown or not rated)
df_rated = df_info_only[df_info_only['IMDb Rating'] != 0]

# Convert 'IMDb Rating' to numeric if needed
df_rated['IMDb Rating'] = pd.to_numeric(df_rated['IMDb Rating'], errors='coerce')

# Display the IMDb ratings distribution
imdb_distribution = df_rated['IMDb Rating'].value_counts().sort_index()

# Show the IMDb ratings distribution
imdb_distribution


<ipython-input-33-bb734d75c94e>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_rated['IMDb Rating'] = pd.to_numeric(df_rated['IMDb Rating'], errors='coerce')


IMDb Rating
2.8     1
2.9     1
3.1     1
3.3     2
3.8     1
4.0     1
4.3     6
4.4     5
4.5     5
4.6     3
4.7     4
4.8     6
4.9     2
5.0     1
5.1     4
5.2     7
5.3    17
5.4    12
5.5    12
5.6    17
5.7    20
5.8    22
5.9    15
6.0    11
6.1    18
6.2    23
6.3    35
6.4    25
6.5    26
6.6    27
6.7    30
6.8    19
6.9    26
7.0    33
7.1    20
7.2    25
7.3    22
7.4    27
7.5    23
7.6    27
7.7    26
7.8    30
7.9    15
8.0    25
8.1    21
8.2    23
8.3    13
8.4    24
8.5    14
8.6     6
8.7    13
8.8     7
8.9     3
9.1     1
9.2     1
Name: count, dtype: int64

## **Ratings by Content Type (Excluding Unknown Content Type and 0 IMDb Rating)**
Analysing whether there’s a difference in IMDb ratings between various content types, such as TV series and movies. This helps us to understand whether certain types of content tend to have higher or lower ratings.

In [34]:
# Exclude rows where 'Content Type' is 'Unknown' and 'IMDb Rating' is 0
df_filtered = df_info_only[(df_info_only['Content Type'] != 'Unknown') & (df_info_only['IMDb Rating'] != 0)]

# Convert 'IMDb Rating' to numeric if needed
df_filtered['IMDb Rating'] = pd.to_numeric(df_filtered['IMDb Rating'], errors='coerce')

# Display the IMDb ratings distribution
imdb_distribution = df_filtered['IMDb Rating'].value_counts().sort_index()

# Show the IMDb ratings distribution
imdb_distribution


<ipython-input-34-963cf105f806>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_filtered['IMDb Rating'] = pd.to_numeric(df_filtered['IMDb Rating'], errors='coerce')


IMDb Rating
2.8     1
2.9     1
3.1     1
3.3     2
3.8     1
4.0     1
4.3     6
4.4     5
4.5     5
4.6     3
4.7     4
4.8     6
4.9     2
5.0     1
5.1     4
5.2     7
5.3    17
5.4    12
5.5    12
5.6    17
5.7    20
5.8    22
5.9    15
6.0    11
6.1    18
6.2    23
6.3    35
6.4    25
6.5    26
6.6    27
6.7    30
6.8    19
6.9    26
7.0    33
7.1    20
7.2    25
7.3    22
7.4    27
7.5    23
7.6    27
7.7    26
7.8    30
7.9    15
8.0    25
8.1    21
8.2    23
8.3    13
8.4    24
8.5    14
8.6     6
8.7    13
8.8     7
8.9     3
9.1     1
9.2     1
Name: count, dtype: int64

In [35]:
# Group by Content Type and calculate the average IMDb rating (excluding 'Unknown' and '0' ratings)
imdb_by_content_type = df_filtered.groupby('Content Type')['IMDb Rating'].mean()

# Display the average IMDb rating by content type
imdb_by_content_type

Content Type
Movie        6.321090
TV Series    7.495288
Name: IMDb Rating, dtype: float64

# **4. Viewing Trends Analysis**

**Summary of Viewing Trends Analysis:**

Unique Shows/Movies per Month: Shows how many distinct shows or movies you watched in each month.

Unique Shows/Movies per Year: Provides a yearly summary of your unique viewing activity.

### **Count of Unique Shows/Movies Watched per Month**

In [36]:
unique_shows_per_month = df_time_series.groupby(['Year', 'Month'])['Title'].nunique()

unique_shows_per_month

Year  Month
2017  8         1
      9        10
      10       14
      11       14
      12       31
               ..
2024  5        13
      6         6
      7        18
      8        10
      9         7
Name: Title, Length: 86, dtype: int64

### **Count of Unique Shows/Movies Watched per Year**

In [37]:
unique_shows_per_year = df_time_series.groupby('Year')['Title'].nunique()

unique_shows_per_year

Year
2017     63
2018    112
2019    105
2020     95
2021    197
2022    105
2023    150
2024     94
Name: Title, dtype: int64

# **Downloading both dataframes**

In [38]:
from google.colab import files

# Save the time series DataFrame (with duplicates and unknowns)
df_time_series.to_csv('netflix_time_series_data.csv', index=False)

# Save the info-only DataFrame (without duplicates and with cleaned data)
df_info_only.to_csv('netflix_cleaned_info_data.csv', index=False)

# Download the CSV files
files.download('netflix_time_series_data.csv')
files.download('netflix_cleaned_info_data.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>